File: backend/app/schemas/company.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List

class JobCreate(BaseModel):
    title: str
    city: Optional[str] = None
    province: Optional[str] = None
    industry: Optional[str] = None
    min_salary: Optional[int] = None
    max_salary: Optional[int] = None
    min_degree: Optional[int] = 1
    min_exp_years: Optional[int] = 0
    keywords: Optional[List[str]] = None
    description: Optional[str] = None


class JobStatusUpdate(BaseModel):
    status: int  # 0=暂停, 1=招聘中, 2=已结束

class JobResponse(BaseModel):
    job_id: str
    company_id: str
    title: str
    city: Optional[str]
    province: Optional[str]
    industry: Optional[str]
    min_salary: Optional[int]
    max_salary: Optional[int]
    min_degree: int
    min_exp_years: int
    keywords: Optional[List[str]]
    description: Optional[str]
    status: int
    published_at: Optional[str]
    created_at: str

class CompanyDashboardResponse(BaseModel):
    published_jobs: int
    received_resumes: int
    hired_count: int
    trend_data: List[dict]


class ProfileUpdateRequest(BaseModel):
    """企业档案更新请求（提交审核）"""
    company_name: str
    industry: Optional[str] = None
    city: Optional[str] = None
    size: Optional[str] = None
    description: Optional[str] = None
    address: Optional[str] = None
    email: Optional[str] = None
    contact: Optional[str] = None
    contact_phone: Optional[str] = None


class ProfilePendingResponse(BaseModel):
    """待审核信息响应"""
    pending_id: str
    company_id: str
    address: Optional[str]
    email: Optional[str]
    contact: Optional[str]
    contact_phone: Optional[str]
    status: str
    reject_reason: Optional[str]
    submitted_at: str
    reviewed_at: Optional[str]

File: src/api/company.ts
import request from '@/utils/http'

export const fetchCompanyDashboard = () => request.get<any>({ url: '/api/v1/company/dashboard' })

export const fetchEnterpriseDataboard = (year?: number) =>
  request.get<any>({ url: '/api/v1/company/enterprise-databoard', params: year ? { year } : undefined })

export const fetchEnterpriseWordCloud = () =>
  request.get<any>({ url: '/api/v1/company/enterprise-databoard/word-cloud' })

export const fetchEnterpriseJobTitles = () =>
  request.get<any>({ url: '/api/v1/company/enterprise-databoard/job-titles' })

export const fetchCompanyJobs = (params?: any) =>
  request.get<any>({ url: '/api/v1/company/jobs', params })

export const getJob = (jobId: string) => request.get<any>({ url: `/api/v1/company/jobs/${jobId}` })

export interface JobCreateParams {
  title: string
  city?: string
  province?: string
  industry?: string
  min_salary?: number
  max_salary?: number
  min_degree?: number
  min_exp_years?: number
  keywords?: string[]
  description?: string
}

export const createJob = (data: JobCreateParams) =>
  request.post<any>({ url: '/api/v1/company/jobs', data })

export const updateJob = (jobId: string, data: JobCreateParams) =>
  request.put<any>({ url: `/api/v1/company/jobs/${jobId}`, data })

export const deleteJob = (jobId: string) =>
  request.del<any>({ url: `/api/v1/company/jobs/${jobId}` })

export const toggleJobStatus = (jobId: string, status: number) =>
  request.patch<any>({ url: `/api/v1/company/jobs/${jobId}/status`, data: { status } })

export const fetchCompanyProfile = () => request.get<any>({ url: '/api/v1/company/profile' })

export const updateCompanyProfile = (data: any) =>
  request.put<any>({ url: '/api/v1/company/profile', data })

export interface ProfileUpdateParams {
  company_name: string
  industry?: string
  city?: string
  size?: string
  description?: string
  address?: string
  email?: string
  contact?: string
  contact_phone?: string
}

export const submitCompanyProfileForReview = (data: ProfileUpdateParams) =>
  request.post<any>({ url: '/api/v1/company/profile/submit', data })

export const fetchPendingProfile = () =>
  request.get<any>({ url: '/api/v1/company/profile/pending' })

// 简历管理
export interface ResumeListParams {
  status?: number
  page?: number
  page_size?: number
}

export const fetchCompanyResumes = (params?: ResumeListParams) =>
  request.get<any>({ url: '/api/v1/company/resumes', params })

export const updateResumeStatus = (applicationId: string, status: number) =>
  request.patch<any>({
    url: `/api/v1/company/resumes/${applicationId}/status`,
    data: { status }
  })

export const fetchStudentByAccount = (accountId: string) =>
  request.get<any>({ url: `/api/v1/company/resumes/student/${accountId}` })
